lib/ui/painting/image_decoder_unittests.cc

```diff
@@ -557,5 +612,87 @@ TEST(ImageDecoderTest, VerifySubpixelDecodingPreservesExifOrientation) {
   assert_image(decode({}, 100));
 }
 
+TEST_F(ImageDecoderFixtureTest,
```

Comment:
This test is pretty big, it can't be groked with a passing perusal.  I recommend adding a docstring at the beginning of the test, or pulling out some helper functions so that the gist can be parsed quicker.

Reply:
Added a comment to the top.  I'm not quite sure how to refactor this into smaller functions while still preserving the ordering that's needed, but maybe when I get in we can talk about it some more offline.